constraints/starCat.js

```diff
@@ -0,0 +1,15 @@
+//Assign a violation for every node of category c in p.
+//Truckebrodt (1995, 1999): *phi
+function starCat(s, p, c){
+	var occurances = 0;
+	if (p.cat === c){
+		occurances ++;
+	};
+	if (p.children){
```

Comment:
change to if (p.children && p.children.length) to be sure to handle the case where p.children has been declared but not initialized.